### PR TITLE
[native] Add documentations to shuffle related codebase

### PR DESCRIPTION
**Pull request by tanjialiang**
The commits in this PR:
1. Add documentations to some recently added shuffle related code.
2. Fix license headers on these recently added files
```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Review by gggrace14 — APPROVED** [review]
It looks good to me. Thank you for adding the docs!

File: presto-native-execution/presto_cpp/main/operators/PartitionAndSerialize.cpp [hunk 1]
```diff
@@ -1,6 +1,4 @@
 /*
- * Copyright (c) Facebook, Inc. and its affiliates.
```
Comment (mbasmanova) [inline presto-native-execution/presto_cpp/main/operators/PartitionAndSerialize.cpp:2]: Accidental change? Run make header-fix to make sure headers are correct.

File: presto-native-execution/presto_cpp/main/operators/PartitionAndSerialize.cpp [hunk 2]
```diff
@@ -45,6 +43,11 @@ void PartitionAndSerializeNode::addDetails(std::stringstream& stream) const {
 
 namespace {
 
+/// Partitions the input row based on partition function and serializes the
+/// entire row.
+/// The output of this operator is a row with 2 columns, with the
```
Comment (mbasmanova) [inline presto-native-execution/presto_cpp/main/operators/PartitionAndSerialize.cpp:48]: > The output of this operator is a row with 2 columns, with the

The output has 2 columns: (1) partition number (INTEGER); (2) serialized row (VARBINARY).

File: presto-native-execution/presto_cpp/main/operators/PartitionAndSerialize.cpp [hunk 3]
```diff
@@ -45,6 +43,11 @@ void PartitionAndSerializeNode::addDetails(std::stringstream& stream) const {
 
 namespace {
 
+/// Partitions the input row based on partition function and serializes the
```
Comment (mbasmanova) [inline presto-native-execution/presto_cpp/main/operators/PartitionAndSerialize.cpp:46]: Nice comment. Let's move it to the plan node class.

File: presto-native-execution/presto_cpp/main/operators/TestingPersistentShuffle.h [hunk 4]
```diff
@@ -66,7 +66,7 @@ class TestingPersistentShuffle : public ShuffleInterface {
     if (rootPath_ != "") {
       cleanup();
     }
-    rootPath_ = std::move(rootPath);
+    rootPath_ = rootPath;
```
Comment (mbasmanova) [inline presto-native-execution/presto_cpp/main/operators/TestingPersistentShuffle.h:69]: Why this change?

File: presto-native-execution/presto_cpp/main/operators/tests/UnsaferowShuffleTest.cpp [hunk 5]
```diff
@@ -78,6 +78,7 @@ class TestShuffle : public ShuffleInterface {
 
   void noMoreData(bool success) {
     VELOX_CHECK(success, "Unexpected error")
+    // Flush out any leftover in progress buffer to ready buffer
```
Comment (mbasmanova) [inline presto-native-execution/presto_cpp/main/operators/tests/UnsaferowShuffleTest.cpp:81]: Add a period at the end of the sentence.

"Flush in-progress buffers."

**Review by mbasmanova — COMMENTED** [review]
@tanjialiang Thank you for adding comments. The changes to the license header don't seem quite right. Please, double check and mention these in the commit message. It would be better to put these in a separate commit.

**Comment by v-jizhang** [comment]
@bot kick off testes

File: presto-native-execution/presto_cpp/main/operators/TestingPersistentShuffle.h [hunk 6]
```diff
@@ -51,7 +51,7 @@ class TestingPersistentShuffle : public ShuffleInterface {
   void initialize(
       velox::memory::MemoryPool* pool,
       uint32_t numPartitions,
-      std::string rootPath) {
+      const std::string& rootPath) {
```
Comment (mbasmanova) [inline presto-native-execution/presto_cpp/main/operators/TestingPersistentShuffle.h:54]: Why this change?
